// react-material-music-player/dist/components/Playlist/Playlist.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Playlist;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("@mui/material/Box"));

var _reactRedux = require("react-redux");

var _actionCreators = _interopRequireDefault(require("../../redux/actionCreators.js"));

var _reactDraggableList = _interopRequireDefault(require("react-draggable-list"));

var _PlaylistItemTemplate = _interopRequireDefault(require("./PlaylistItemTemplate.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function Playlist(props) {
  var sx = props.sx;

  var _useSelector = (0, _reactRedux.useSelector)(
  /** @type {import("../../redux/types.js").useSelectCb} */
  function (_ref) {
    var playlist = _ref.playlist,
        currentTrack = _ref.currentTrack;
    return {
      playlist: playlist,
      currentTrack: currentTrack
    };
  }, _reactRedux.shallowEqual),
      playlist = _useSelector.playlist,
      currentTrack = _useSelector.currentTrack;

  var dispatch = (0, _reactRedux.useDispatch)();

  var onReorder = function onReorder(newList) {
    return dispatch(_actionCreators.default.updatePlaylist(newList));
  };

  var onTrackSelect = function onTrackSelect(index) {
    // change and play track immediately
    dispatch(_actionCreators.default.changeTrack(index));
    dispatch(_actionCreators.default.play());
  };

  var draggablelistContainerRef = /*#__PURE__*/_react.default.createRef();

  return /*#__PURE__*/_react.default.createElement(_Box.default, {
    ref: draggablelistContainerRef,
    sx: _objectSpread({
      margin: function margin(theme) {
        return theme.spacing();
      },
      // require set container with fix size for draggable list
      width: "10vw",
      height: "10vh"
    }, sx)
  }, /*#__PURE__*/_react.default.createElement(_reactDraggableList.default, {
    list: playlist,
    itemKey: "ID",
    template: _PlaylistItemTemplate.default,
    onMoveEnd: onReorder,
    container: function container() {
      return draggablelistContainerRef.current;
    },
    commonProps: {
      listOfID: playlist.map(function (element) {
        return element.ID;
      }),
      currentTrackID: playlist[currentTrack].ID,
      onTrackSelect: onTrackSelect
    }
  }));
}
