// react-material-music-player/dist/components/Controls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Controls;

var _Box2 = _interopRequireDefault(require("@mui/material/Box"));

var _react = _interopRequireDefault(require("react"));

var _reactRedux = require("react-redux");

var _IconButton = _interopRequireDefault(require("@mui/material/IconButton"));

var _SkipNextRounded = _interopRequireDefault(require("@mui/icons-material/SkipNextRounded"));

var _SkipPreviousRounded = _interopRequireDefault(require("@mui/icons-material/SkipPreviousRounded"));

var _PlayArrowRounded = _interopRequireDefault(require("@mui/icons-material/PlayArrowRounded"));

var _PauseRounded = _interopRequireDefault(require("@mui/icons-material/PauseRounded"));

var _actionCreators = _interopRequireDefault(require("../redux/actionCreators"));

var _withoutPropagation = _interopRequireDefault(require("../utils/withoutPropagation"));

var _types = require("../redux/types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function Controls(props) {
  var sx = props.sx;
  var mediaState = (0, _reactRedux.useSelector)(
  /** @type {import("../redux/types").useSelectCb} */
  function (_ref) {
    var mediaState = _ref.mediaState;
    return mediaState;
  });
  var dispatch = (0, _reactRedux.useDispatch)();

  var onSkipNext = function onSkipNext() {
    return dispatch(_actionCreators.default.skipNext());
  };

  var onSkipPrev = function onSkipPrev() {
    return dispatch(_actionCreators.default.skipPrev());
  };

  var onPlay = function onPlay() {
    return dispatch(_actionCreators.default.play());
  };

  var onPause = function onPause() {
    return dispatch(_actionCreators.default.pause());
  };

  var playing = mediaState === _types.MediaState.PLAYING ? true : false;
  return /*#__PURE__*/_react.default.createElement(_Box2.default, {
    sx: _objectSpread({
      display: "flex",
      flexDirection: "row",
      justifyContent: "center",
      alignItems: "center",
      flexWrap: "nowrap"
    }, sx)
  }, /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    onClick: (0, _withoutPropagation.default)(onSkipPrev),
    size: "large"
  }, /*#__PURE__*/_react.default.createElement(_SkipPreviousRounded.default, {
    fontSize: "large"
  })), /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    onClick: (0, _withoutPropagation.default)(playing ? onPause : onPlay),
    size: "large"
  }, playing ? /*#__PURE__*/_react.default.createElement(_PauseRounded.default, {
    fontSize: "large"
  }) : /*#__PURE__*/_react.default.createElement(_PlayArrowRounded.default, {
    fontSize: "large"
  })), /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    onClick: (0, _withoutPropagation.default)(onSkipNext),
    size: "large"
  }, /*#__PURE__*/_react.default.createElement(_SkipNextRounded.default, {
    fontSize: "large"
  })));
}
