// react-material-music-player/dist/components/ProgressBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ProgressBar;

var _Box2 = _interopRequireDefault(require("@mui/material/Box"));

var _Slider2 = _interopRequireDefault(require("@mui/material/Slider"));

var _react = _interopRequireDefault(require("react"));

var _reactRedux = require("react-redux");

var _actionCreators = _interopRequireDefault(require("../redux/actionCreators.js"));

var _Typography = _interopRequireDefault(require("@mui/material/Typography"));

var _secondsToString = _interopRequireDefault(require("../utils/secondsToString.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function ProgressBar(props) {
  var sx = props.sx;

  var _useSelector = (0, _reactRedux.useSelector)(
  /**@type {import("../redux/types.js").useSelectCb} */
  function (_ref) {
    var timeLeft = _ref.timeLeft,
        currentTime = _ref.currentTime;
    return {
      timeLeft: timeLeft,
      currentTime: currentTime
    };
  }, _reactRedux.shallowEqual),
      timeLeft = _useSelector.timeLeft,
      currentTime = _useSelector.currentTime; // NaN on division by zero


  var progress = currentTime / (timeLeft + currentTime) * 100 || 0;
  var dispatch = (0, _reactRedux.useDispatch)();

  var onSeek = function onSeek(progress) {
    return dispatch(_actionCreators.default.seek(progress));
  };

  var handleSliderChange = function handleSliderChange(event, newValue) {
    onSeek(newValue);
  };

  return /*#__PURE__*/_react.default.createElement(_Box2.default, {
    sx: _objectSpread({
      display: "flex",
      flexDirection: "row",
      wrap: "nowrap",
      alignItems: "center",
      "& > .children": {
        mx: 1
      }
    }, sx)
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: "children"
  }, (0, _secondsToString.default)(currentTime)), /*#__PURE__*/_react.default.createElement(_Slider2.default, {
    className: "children",
    "aria-labelledby": "continuous-slider",
    value: progress,
    onChange: handleSliderChange
  }), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: "children"
  }, (0, _secondsToString.default)(timeLeft)));
}
